### PR TITLE
Write script_json after rubric update

### PR DESCRIPTION
**Pull request by bethanyaconnor**
tl;dr this PR:
1. writes script_json after update
2. sdds a new factory trait to create a lesson group automatically (I looked at doing this without the trait but hit a trail of failures that I don't have the cycles to track down right now)
3. makes a small change to help with the changing order of evidence levels

This one was a bit of an adventure. At first it seemed like a quick fix to simply add one line to the controller (done in 91901dd6c98808d17304d01ecf41851b2e0fe7a5). However, then I went to test it and could not get the tests to pass. Add to that that I wasn't always seeing my changes show up when manually saving and I went on a bit of a goose chase.

In the end, I can no longer repro not seeing the changes show up. Two possibilities for this (other than a magical fix) are either that, because of thrashing in the script_json file, I missed the change or I simply thought I was on a different branch. (Possibly both at different times.) Either way, I am putting down that mystery now and I've included a fix for the thrashing.

The test was its own rabbit hole. Long story short, it turns out that lessons need to be in a lesson group to be serialized, which doesn't happen by default. Once I finally realized that, my tests passed.

### CODE REVIEW

**Review by davidsbailey — APPROVED** [review]
This looks like quite the debugging adventure followed by a great choice of solution and stopping point. I agree with not further trying to debug why it seemed like it wasn't working earlier :-) very nice sleuthing! at the very least if whatever problem was cropping up earlier shows up again, you'll have made it much easier to debug next time.
